haskell-for-readers.md

```diff
@@ -2670,8 +2687,27 @@ What can we do with that freedom? Create more instances!
   ```
   is not even a `Functor`: Since a `Printer a` *expects* a value of type `a`, you can’t apply a function `a -> b` anywhere. (You could apply a function `b -> a`; this means that `Printer` is a *contravariant functor*.)
 
-At a high level, the main advantage of an applicative computation over a monadic is that the *structure of the computation* is static, and can be known ahead of times. This is not the case in a monadic computation, because the second argument to `(>>=)` is an opaque function, and only once a concrete `a` was extracted from the first argument do we know what kind of computation is produced by the second.
+At a high level, the main advantage of an applicative computation over a monadic one is that the *structure of the computation* is static, and can be known ahead of time. This is not the case in a monadic computation, because the second argument to `(>>=)` is an opaque function, and only once a concrete `a` was extracted from the first argument do we know what kind of computation is produced by the second.
 
 As a concrete example, consider a library that provides a type constructor `AParser` that is an instance of `Applicative`, but *not* of `Monad`. You write a parser `p :: AParser a` using this interface. Then the library can “run” this parser without actual input, and learn everything it *would* do. It could use that to provide a function `describe :: AParser a -> String` that produces a grammar for the given parser, which you can use as documentation, and thus these can never go out of sync.
 
 
+{::comment}
+Solution 16
+  p59: ... and dedicated data types with more *speaking* names ...
+       [In German "sprechend", like in Dutch "sprekend", would be fine, but
+        I think for English something like "descriptive" would be better.]
+Solution 23
+  p60: 1. ... that are more likely to produce a balance*d* tree ...
+       2. ... to combine values that are present in bo*th* trees:
+
+
+[On pages 61 and 62 the order of the solutions is: 24, 27, 25, 26, 28.]
+
+
+Solution 26
+  p62: "m" sounds like it starts with a vowel, so both "a m a" should be
+       *an m a*.
+Solution 28
+  p62: ... exactly the operations needed to [*-to*] implement pure ...
+{:/comment}
```

Comment:
The solutions are actually also part of this `.md` files, so you can change them directly.